src/components/PopupWithForm.js

```diff
@@ -5,6 +5,19 @@ class PopupWithForm extends Popup {
     super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector(".modal__form");
+    console.log(this._popupElement); // Check if this logs the correct element
+    if (!this._popupElement) {
+      throw new Error(`Element not found for selector: ${selector}`);
+    }
+  }
+
+  setLoading(isLoading) {
+    this._submitBtn = this._popupElement.querySelector(".modal__button");
+    if (isLoading) {
+      this._submitBtn.textContent = "Saving...";
+    } else {
+      this._submitBtn.textContent = "Submit";
+    }
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`